testing/src/main/java/io/split/client/testing/cucumber/CucumberSplit.java

```diff
@@ -0,0 +1,51 @@
+package io.split.client.testing.cucumber;
+
+import io.cucumber.java.Scenario;
+import io.split.client.testing.SplitClientForTest;
+
+import java.util.Collection;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * <p>
+ * Simple Cucumber plugin for Split.
+ * </p>
+ * <p>
+ * Cucumber scenarios annotated with {@code @split[feature:treatment]} tags can be used to
+ * configure a {@link SplitClientForTest} instance.
+ * </p>
+ * <p>
+ * To use it, define a <a href="https://cucumber.io/docs/cucumber/api/#hooks">Before Hook</a> that invokes the {@link CucumberSplit#configureSplit(SplitClientForTest, Scenario)}
+ * method. Example:
+ * </p>
+ *
+ * <pre>
+ * import io.cucumber.java.Before;
+ * import io.split.client.testing.SplitClientForTest;
+ *
+ * public class StepDefinitions {
+ *     private final SplitClientForTest splitClient = new SplitClientForTest();
+ *
+ *     &#64;Before
+ *     public void configureSplit(Scenario scenario) {
+ *         CucumberSplit.configureSplit(splitClient, scenario);
+ *     }
+ * }
+ * </pre>
+ */
+public class CucumberSplit {
+    private static final Pattern SPLIT_TAG_PATTERN = Pattern.compile("^@split\\[(.*):(.*)]");
+
+    public static void configureSplit(SplitClientForTest splitClient, Scenario scenario) {
+        Collection<String> tags = scenario.getSourceTagNames();
+        for (String tag : tags) {
+            Matcher matcher = SPLIT_TAG_PATTERN.matcher(tag);
```

Comment:
Question here:
- if more than 1 tag matches this, do you end up overwriting the registered (feature, treatment) pair?

Reply:
Yes. This is intentional.

Tags that are placed at the top level `Feature` level are inherited by all the `Scenario`s. The `getSourceTagNames()` method returns a `Collection` of the tags from both the `Feature` and the current `Scenario`. The inherited `Feature` tags are always before the `Scenario` tags in the collection (it would have been more appropriate to declare the return type as `List` rather than `Collection` since it's ordered).

We're taking advantage of this here to allow users to define "default" treatments at the feature level, and (optionally) override them at the scenario level. I have illustrated this in the last scenario where the default `dollars=off` treatment from the feature level is overridden to `dollars=on` for that particular scenario.

Reply:
I've pushed some more changes to (hopefully) illustrate better how this works.